src/storage-client/src/controller.rs

```diff
@@ -1856,19 +1900,28 @@ mod persist_read_handles {
 
                         futs.push(async move {
                             let epoch = since_handle.opaque().clone();
-                            let result = since_handle
-                                .maybe_compare_and_downgrade_since(&epoch, (&epoch, &since))
-                                .instrument(span)
-                                .await;
-                            if let Some(Err(other_epoch)) = result {
+                            let result = if since.is_empty() {
```

Comment:
Might be worth to call this out in a comment: "We want to make sure that advancement to the empty frontier ([]) is guaranteed, which we don't get when doing `maybe_compare_and_downgrade`.